File: go-backend/internal/model/entity/recharge_record.go
package entity

import "time"

type RechargeRecord struct {
	ID            int64     `json:"id,string" gorm:"column:id;primaryKey;autoIncrement"`
	UserID        int64     `json:"userId,string" gorm:"column:userId"`
	Amount        float64   `json:"amount" gorm:"column:amount"`
	PaymentMethod string    `json:"paymentMethod" gorm:"column:paymentMethod"`
	PaymentID     string    `json:"paymentId" gorm:"column:paymentId"`
	Status        string    `json:"status" gorm:"column:status"`
	Description   string    `json:"description" gorm:"column:description"`
	CreateTime    time.Time `json:"createTime" gorm:"column:createTime"`
	UpdateTime    time.Time `json:"updateTime" gorm:"column:updateTime"`
}

func (RechargeRecord) TableName() string {
	return "recharge_record"
}

File: python-backend/app/models/recharge_record.py
"""Recharge record model."""

from __future__ import annotations

from datetime import datetime
from decimal import Decimal

from sqlalchemy import BigInteger, DateTime, Index, Numeric, String, func, text
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class RechargeRecord(Base):
    __tablename__ = "recharge_record"
    __table_args__ = (
        Index("idx_paymentId", "paymentId"),
        Index("idx_status", "status"),
        Index("idx_userId", "userId"),
    )

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    user_id: Mapped[int] = mapped_column("userId", BigInteger, nullable=False)
    amount: Mapped[Decimal] = mapped_column(Numeric(12, 4), nullable=False)
    payment_method: Mapped[str] = mapped_column("paymentMethod", String(32), nullable=False)
    payment_id: Mapped[str | None] = mapped_column("paymentId", String(256), nullable=True)
    status: Mapped[str] = mapped_column(
        String(32),
        nullable=False,
        default="pending",
        server_default=text("'pending'"),
    )
    description: Mapped[str | None] = mapped_column(String(512), nullable=True)
    create_time: Mapped[datetime] = mapped_column(
        "createTime",
        DateTime,
        nullable=False,
        default=datetime.utcnow,
        server_default=func.current_timestamp(),
    )
    update_time: Mapped[datetime] = mapped_column(
        "updateTime",
        DateTime,
        nullable=False,
        default=datetime.utcnow,
        onupdate=datetime.utcnow,
        server_default=func.current_timestamp(),
    )
